### PR TITLE
Validation of parameters is deprecated

### PR DESCRIPTION
**Pull request by cabrinha**
Basically, when using this module, I get this error in my puppetserver logs. 

It seems that the validate_re function has been deprecated recently and I'd like to update this module to adhere to this.

https://github.com/puppetlabs/puppetlabs-stdlib#validate_re

Error:
```
2019-03-06T20:41:33.752Z WARN  [qtp1665900938-77] [puppetserver] Puppet This method is deprecated, please use the stdlib validate_legacy function,
                    with Pattern[]. There is further documentation for validate_legacy function in the README. at ["/etc/puppetlabs/code/environments/master/modules/nslcd/manifests/init.pp", 39]:["/etc/puppetlabs/code/environments/master/site/profile/manifests/ldap.pp", 5]
```

Unrelated, but your travis builds are also failing @geekix, I think the fix for that may be here:

https://github.com/rubygems/rubygems/issues/2534#issuecomment-448843746

### CODE REVIEW

**Comment by geekix** [comment]
Hello @cabrinha ,

Thanks for the PR, I will look into this today.

**Comment by hdep** [comment]
Hi I think we should use puppet datatype for variable https://puppet.com/docs/puppet/4.9/lang_data_type.html

**Comment by geekix** [comment]
@cabrinha For my information, which Puppet version are you using ?

**Comment by cabrinha** [comment]
@geekix I'm currently on Puppet 6, but I was also seeing these errors with Puppet 4.10.x

**Comment by geekix** [comment]
Hello,

@cabrinha I found no easy way to update your PR, so can you please add the data types ?

Thanks

**Comment by cabrinha** [comment]
@geekix no problem, I've gone ahead and added types for each param.

File: manifests/init.pp [hunk 1]
```diff
@@ -2,45 +2,34 @@
...
-  $ldap_search_base    = $nslcd::params::ldap_search_base,
-  $ldap_group_base    =  $nslcd::params::ldap_group_base,
-  $ldap_search_scope   = $nslcd::params::ldap_search_scope,
-  $config_options      = $nslcd::params::config_options,
-  $ldap_filters        = $nslcd::params::ldap_filters,
-  $ldap_maps           = $nslcd::params::ldap_maps,
-  $ldap_ssl            = $nslcd::params::ldap_ssl,
-  $ldap_tls_reqcert    = $nslcd::params::ldap_tls_reqcert,
-  $ldap_tls_cacertfile = $nslcd::params::ldap_tls_cacertfile,
+  Variant[Boolean, Enum['true', 'false', 'present', 'absent']] $package_ensure  = $nslcd::params::package_ensure,
```
Comment (natemccurdy) [inline manifests/init.pp]: May be easier to just set this type to be `Variant[Boolean,String[1]]`.
Some folks might want to set a version number for their package, and Enum won't allow that.

Comment (cabrinha) [reply manifests/init.pp]: Done

**Comment by cabrinha** [comment]
I may end up closing this PR in favor of: https://github.com/natemccurdy/puppet-nslcd/tree/param_types

**Comment by geekix** [comment]
Hello,

@cabrinha thanks for the PR update. The @natemccurdy fork code is quite good indeed, but I also reviewed your code and it's pretty decent as well ^^. Your call, what's your advice ?

**Comment by natemccurdy** [comment]
@cabrinha I've put up a PR against my branch if you'd prefer to go with it over yours.

**Comment by cabrinha** [comment]
I’ll close this in favor of Nate’s.

**Pull request closed without merging**